actors/miner/src/lib.rs

```diff
@@ -3818,23 +3731,62 @@ fn extend_sector_committment_legacy(
         ));
     }
 
-    validate_expiration2(policy, curr_epoch, sector.activation, new_expiration, sector.seal_proof)?;
+    validate_expiration(policy, curr_epoch, sector.activation, new_expiration, sector.seal_proof)?;
+    Ok(())
+}
 
-    // Remove "spent" deal weights
+fn extend_simple_qap_sector(
+    new_expiration: ChainEpoch,
+    sector: &SectorOnChainInfo,
+    claim_space_by_sector: &BTreeMap<SectorNumber, u64>,
+) -> Result<SectorOnChainInfo, ActorError> {
+    let mut new_sector = sector.clone();
+    if sector.verified_deal_weight > BigInt::zero() {
+        let old_duration = sector.expiration - sector.activation;
+        let deal_space = &sector.deal_weight / old_duration;
+        let verified_deal_space = &sector.verified_deal_weight / old_duration;
+        let expected_verified_deal_space = match claim_space_by_sector.get(&sector.sector_number) {
+            None => {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "claim missing from declaration for sector {}",
+                    sector.sector_number
+                ))
+            }
+            Some(space) => space,
+        };
+        if BigInt::from(*expected_verified_deal_space as i64) != verified_deal_space {
+            return Err(actor_error!(illegal_argument, "declared verified deal space in claims ({}) does not match verified deal space ({}) for sector {}", expected_verified_deal_space, verified_deal_space, sector.sector_number));
+        }
+        new_sector.expiration = new_expiration;
+        // update deal weights to account for new duration
+        new_sector.deal_weight = deal_space * (new_sector.expiration - new_sector.activation);
+        new_sector.verified_deal_weight =
+            verified_deal_space * (new_sector.expiration - new_sector.activation);
+    } else {
+        new_sector.expiration = new_expiration
```

Comment:
We weren't extending non QAP sectors before, refactoring helped me catch this bug 🙏

Reply:
I thought about this but then we don't have the nice property that duration is specified in terms of fields on the sector info (new_sector.expiration - new_sector.activation) which you asked for in the past and I agree makes this tricky detail very explicit.